content/docs/guides/query-log.md

```diff
@@ -0,0 +1,182 @@
+---
+title: Query Log
+sort_rank: 1
+---
+
+# Using the Prometheus query log
+
+Prometheus has the ability to log all the queries run by the engine to a log
+file. This guide demonstrates how to use that log file, which fields it does
+contain and provides advanced tips about how to operate the log file.
+
+The query log has been introduced in v2.16.0 and is disabled by default.
+
+## Enable the query log
+
+The query log can be enabled and disabled at runtime. It can therefore be
+activated when you want to investigate slownesses or high load on your
+Prometheus instance.
+
+To enable or disable the query log, two steps are needed:
+
+1. Adapt the configuration to add or remove the query log configuration.
+1. Reload the Prometheus server configuration.
+
+### Logging all the queries to a file
+
+This example demonstrates how to log all the queries to
+a file called `/var/log/prometheus/query.log`.
+
+First, let's ensure that Prometheus has write access to a directory
+`/var/log/prometheus`. We will assume that Prometheus is running under a user
+called Prometheus:
+
+```shell
+# mkdir /var/log/prometheus
```

Comment:
This fails to mention the need for root. Let's keep to examples that don't require root unless it's strictly required.